client/internal/peer/conn.go

```diff
@@ -343,6 +387,22 @@ func (conn *Conn) cleanup() error {
 
 	conn.status = StatusDisconnected
 
+	peerState, err := conn.statusRecorder.GetPeerStatus(conn.config.Key)
```

Comment:
1. You are modifying and existing state out of the lock of StatusRecorder. I think it makes sense to create a copy of PeerStatus

Reply:
we were creating this copy when returning, but it wasn't safe for future use case where there could be a pointer in its reference.

Reply:
the methods was removed